### PR TITLE
docs(tour): added tour documentation

### PR DESCRIPTION
**Pull request by anuraghazra**
### Summary by CodeRabbit

- New Feature: Added support for native platforms in the `Tour` component by introducing `Tour.native.tsx` and `TourStep.native.tsx` files.
- New Feature: Enhanced the functionality and user experience of the `Tour` component in React applications. Changes include improved state management, step navigation, scroll behavior, and smoother transitions between steps.
- New Feature: Added a new utility file `utils.ts` that exports the `useDelayedState` hook and `scrollTo` function for delayed state updates and smooth scrolling.
- Refactor: Uncommented previously disabled code in `preview.js` to enable rendering of stories within the BladeProvider and ErrorBoundary components.
- Enhancement: Added a new function `goToStep` to the `Tour` component, allowing navigation to a specific step in the tour. Modified dependencies of the `useIsomorphicLayoutEffect` hook.
- Enhancement: Modified the `TourProps` interface to make `onOpenChange` and `onFinish` callbacks optional.
- Enhancement: Modified the `TourStep` component by importing and wrapping it with the `assignWithoutSideEffects` function.
- Documentation: Added a new file `Tour.tsx` for rendering a collection of stories using Storybook's `composeStories` function.

### CODE REVIEW

File: packages/blade/src/components/Tour/Tour.stories.tsx [hunk 1]
```diff
@@ -132,151 +315,740 @@ const TourTemplate = () => {
...
+                Step 2
+              </Box>
+            </TourStep>
+          </Box>
+        </Center>
+      </Tour>
+    </Box>
+  );
+};
+InterruptibleTour.storyName = 'Product Usecase: Interruptible Tour';
```
Comment (chaitanyadeorukhkar) [inline packages/blade/src/components/Tour/Tour.stories.tsx:1054]: Just call it Interruptible tour
```suggestion
InterruptibleTour.storyName = 'Interruptible Tour';
```

**Comment by chaitanyadeorukhkar** [comment]
@anuraghazra lets add kitchen sync as well?

File: packages/blade/src/components/Tour/Tour.stories.tsx (hunk 1 above)
Comment (anuraghazra) [reply packages/blade/src/components/Tour/Tour.stories.tsx:1080]: This is an existing product [usecase](https://github.com/razorpay/dashboard/blob/44a954660b0d851cd5fedf48b44d85731e5c48ea/web/js/merchantLA/containers/MerchantTour/index.js#L123-L138) demo. 

I like to prefix the "Product Usecase" to convey this is not a "prop" which can be switched on or off, but rather a whole usecase that consumers create by composing various blade components.

**Comment by anuraghazra** [comment]
> @anuraghazra lets add kitchen sync as well?


Added

**Pull request merged**